salt_lsp/__main__.py

```diff
@@ -43,11 +43,15 @@ def add_arguments(parser):
         help="initialize the server, but don't launch it "
         "(useful for debugging/testing purposes)",
     )
+    parser.add_argument(
+        "--log-file",
+        help="Redirect logs to the given file instead of writing to stderr",
+    )
```

Comment:
```suggestion
    parser.add_argument(
        "--log-file",
        help="Redirect logs to the given file instead of writing to stderr",
        nargs=1,
        type=str,
    )
```

Reply:
`type`: Hmm checking the docs for `type`, it looks like it defaults to string, which agrees with what I see while testing. So we _should_ be ok there, however we can add it to be more explicit if we want?

From https://docs.python.org/3/library/argparse.html#type
> By default, the parser reads command-line arguments in as simple strings. However, quite often the command-line string should instead be interpreted as another type, such as a float or int.


`nargs`: Ahh, I recall having trouble with `nargs` in the past since it can be a little confusing.  Looks like using `nargs=1` would convert it to a list, which would require the `[0]` usage later on.  

From https://docs.python.org/3/library/argparse.html#nargs
> The supported values are:
> * N (an integer). N arguments from the command line will be gathered together into a list. 
> ...
> Note that nargs=1 produces a list of one item. This is different from the default, in which the item is produced by itself.

It might be more clear to leave the `nargs=1` off, so we don't need the `[0]` later?